akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala

```diff
@@ -252,7 +258,8 @@ class LightArrayRevolverScheduler(config: Config,
 
     @tailrec final def nextTick(): Unit = {
       val time = clock()
-      val sleepTime = start + (tick * tickNanos) - time
+      val t = if (tick >= 0) tick else Int.MinValue + tick
+      val sleepTime = start + ((tickWrap + t) * tickNanos) - time
```

Comment:
btw, can this tick tracking logic be extracted into a class that can be tested in isolation? It might be easier to document and test it than together with the whole timer-wheel+spinning stuff.